internal/provider/resource_source.go

```diff
@@ -165,6 +165,28 @@ var sourceSchema = map[string]*schema.Schema{
 		Type:        schema.TypeInt,
 		Optional:    true,
 	},
+	"scrape_request_headers": {
+		Description: "An array of request headers, each containing `name` and `value` fields.",
+		Type:        schema.TypeList,
+		Optional:    true,
+		Elem: &schema.Schema{
+			Type: schema.TypeMap,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+	},
```

Comment:
I can see the headers are loaded as a map, but we only expect `value` and `name`. Users had issues with that in Uptime provider, intuitively filling the map as eg. `{"Bearer": "mytoken"}` which then failed on API in the apply phase.

Could you please add validation for the map and test it in the `resource_source_test.go`? 🙏 

We're doing that in Uptime:
Validation: https://github.com/BetterStackHQ/terraform-provider-better-uptime/blob/6e5c3466129935ee5b98f1ba163fa534d460dcc1/internal/provider/resource_monitor.go#L610
Testing: https://github.com/BetterStackHQ/terraform-provider-better-uptime/blob/6e5c3466129935ee5b98f1ba163fa534d460dcc1/internal/provider/resource_monitor_test.go#L222